framework/db/ConstraintFinderTrait.php

```diff
@@ -0,0 +1,217 @@
+ * array element is an array of [[ForeignKeyConstraint]] or its child classes. This property is read-only.
+ * @property IndexConstraint[][] $schemaIndexes Indexes for all tables in the database. Each array element is
+ * an array of [[IndexConstraint]] or its child classes. This property is read-only.
+ * @property Constraint[] $schemaPrimaryKeys Primary keys for all tables in the database. Each array element
+ * is an instance of [[Constraint]] or its child class. This property is read-only.
+ * @property IndexConstraint[][] $schemaUniques Unique constraints for all tables in the database.
+ * Each array element is an array of [[IndexConstraint]] or its child classes. This property is read-only.
+ *
+ * @author Sergey Makinen <sergey@makinen.ru>
+ * @since 2.0.13
+ */
+trait ConstraintFinderTrait
+{
+    /**
+     * Loads a primary key for the given table.
+     * @param string $tableName table name.
+     * @return Constraint|null primary key for the given table, `null` if the table has no primary key.
+     */
+    abstract protected function loadTablePrimaryKey($tableName);
+
+    /**
+     * Loads all foreign keys for the given table.
+     * @param string $tableName table name.
+     * @return ForeignKeyConstraint[] foreign keys for the given table.
+     */
+    abstract protected function loadTableForeignKeys($tableName);
+
+    /**
+     * Loads all indexes for the given table.
+     * @param string $tableName table name.
+     * @return IndexConstraint[] indexes for the given table.
+     */
+    abstract protected function loadTableIndexes($tableName);
+
+    /**
+     * Loads all unique constraints for the given table.
+     * @param string $tableName table name.
+     * @return Constraint[] unique constraints for the given table.
+     */
+    abstract protected function loadTableUniques($tableName);
+
+    /**
+     * Loads all check constraints for the given table.
+     * @param string $tableName table name.
+     * @return CheckConstraint[] check constraints for the given table.
+     */
+    abstract protected function loadTableChecks($tableName);
+
+    /**
+     * Loads all default value constraints for the given table.
+     *
+     * @param string $tableName table name.
+     * @return DefaultValueConstraint[] default value constraints for the given table.
+     */
+    abstract protected function loadTableDefaultValues($tableName);
+
+    /**
+     * Obtains the primary key for the named table.
+     * @param string $name table name. The table name may contain schema name if any. Do not quote the table name.
+     * @param bool $refresh whether to reload the information even if it is found in the cache.
+     * @return Constraint|null table primary key, `null` if the table has no primary key.
+     */
+    public function getTablePrimaryKey($name, $refresh = false)
+    {
+        return $this->getTableMetadata($name, 'primaryKey', $refresh);
```

Comment:
Sounds reasonable but let's decide on merging `ConstraintFinderTrait` with `Schema`. Maybe these problems will be auto solved.